src/pages/Profile.js

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect } from 'react';
+import styles from './Profile.module.css';
+import { Button as YaButton, Input } from "@ya.praktikum/react-developer-burger-ui-components";
+import { useDispatch, useSelector } from "react-redux";
+import { NavLink, useHistory } from "react-router-dom";
+import Button from "../components/Button/Button";
+import { getUser, logout } from "../services/actions/profile";
+import { setUserForm, userForm } from "../services/actions/userDataForm";
+
+function Profile() {
+  const dispatch = useDispatch();
+
+  const { userData } = useSelector(({ profile }) => profile);
+  const { form } = useSelector(({ userFormReducer }) => userFormReducer);
+  const history = useHistory();
+
+  useEffect
+  (() => {
+    dispatch(getUser());
+  }, [getUser])
```

Comment:
Реакт выдаёт предупреждение: "React Hook useEffect has a missing dependency: 'dispatch'. Either include it or remove the dependency array.
Outer scope values like 'getUser' aren't valid dependencies because mutating them doesn't re-render the component."
Если включить dispatch в зависимости, но не убрать getUser, тогда реакт выдаст предупреждение: "React Hook useEffect has an unnecessary
  dependency: 'getUser'. Either exclude it or remove the dependency array. Outer scope values like 'getUser' aren't valid dependencies because
  mutating them doesn't re-render the component." То есть, когда Вы не передаёте getUser как контекст, она вообще не влияет на перерисовку
  компоненты Profile (то есть постоянный вызов Profile как функции), что, я думаю, очень хорошо.